CRM/ManualDirectDebit/ScheduleJob/TargetContribution.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * Gets target contribution data
+ */
+class CRM_ManualDirectDebit_ScheduleJob_TargetContribution {
+
+  /**
+   * Gets target contribution data
+   *
+   * @return array
+   */
+  public static function retrieve() {
+    $reminderOffsetDays = CRM_ManualDirectDebit_ScheduleJob_ReminderOffsetDays::retrieve();
+    $pendingStatusId = CRM_ManualDirectDebit_Common_OptionValue::getOptionValueID('contribution_status', 'Pending');
+    $cancelledStatusId = CRM_ManualDirectDebit_Common_OptionValue::getOptionValueID('contribution_status', 'Cancelled');
+    $directDebitPaymentInstrumentId = CRM_ManualDirectDebit_Common_OptionValue::getOptionValueID('payment_instrument', 'direct_debit');
+
+    $query = "
+      SELECT 
+        contribution.id AS contribution_id,
+        contact.id AS contact_id,
+        (
+          SELECT email.email
+          FROM civicrm_email AS email
+          WHERE email.contact_id = contact.id
+          LIMIT 1
+        ) AS email
+      FROM civicrm_contribution AS contribution
+      LEFT JOIN civicrm_contact AS contact
+        ON contribution.contact_id = contact.id
+      WHERE 
+        contribution.receipt_date IS NULL
+        AND contribution.payment_instrument_id = %2
+        AND (contribution.contribution_status_id = %3 OR contribution.contribution_status_id = %4)
+        AND contribution.receive_date <= (now() - INTERVAL %1 DAY)
+    ";
```

Comment:
Two things to have into account:

- First, we should get the primary e-mail for the contact.
- Second, we should take into account if the contact has opted out of e-mails, bearing in mind there are two flags to opt-out, so I don't know if one of both should be evaluated (opt_out vs no_email). You may want to run this one by Guanhuan to verify if and which of the flags should be read.